```bash
mypy.....................................................................Failed
- hook id: mypy
- duration: 80.69s
- exit code: 1

pandas/core/arrays/string_.py:841: error: Incompatible types in assignment (expression has type "BooleanArray", variable has type "ndarray[Any, dtype[Any]]")  [assignment]
pandas/core/arrays/string_.py:844: error: "ndarray[Any, dtype[Any]]" has no attribute "to_numpy"  [attr-defined]
pandas/core/arrays/string_.py:846: error: "ndarray[Any, dtype[Any]]" has no attribute "to_numpy"  [attr-defined]
Found 3 errors in 1 file (checked 1441 source files)
```

--- a/pandas/core/arrays/string_.py
+++ b/pandas/core/arrays/string_.py
@@ -838,13 +838,13 @@ class StringArray(BaseStringArray, NumpyExtensionArray):  # type: ignore[misc]
             # logical
             result = np.zeros(len(self._ndarray), dtype="bool")
             result[valid] = op(self._ndarray[valid], other)
-            result = BooleanArray(result, mask)
+            res_arr = BooleanArray(result, mask)
             if self.dtype.na_value is np.nan:
                 if op == operator.ne:
-                    return result.to_numpy(np.bool_, na_value=True)
+                    return res_arr.to_numpy(np.bool_, na_value=True)
                 else:
-                    return result.to_numpy(np.bool_, na_value=False)
-            return result
+                    return res_arr.to_numpy(np.bool_, na_value=False)
+            return res_arr
 
     _arith_method = _cmp_method
 

